File: king-admin-java/src/main/java/com/oukingtim/web/vm/MenuVM.java
/**
 * Project Name:km-ms
 * File Name:SerieTreeVo.java
 * Package Name:com.kingmed.ms.sales.vo
 * Date:2015年5月7日下午4:10:25
 * Copyright (c) 2015, chenzhou1025@126.com All Rights Reserved.
 *
*/

package com.oukingtim.web.vm;

import com.oukingtim.domain.SysMenu;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.beans.BeanUtils;

import java.io.Serializable;
import java.util.List;

/**
 * 菜单数据返回对象
 * Created by oukingtim
 */
@Data
@EqualsAndHashCode(callSuper = false)
public class MenuVM implements Serializable {

    private String id;
    private String parentId;
    private String name;
    private String stateRef;
    private String type;
    private String icon;
    private String title;
    private Integer level;
    private Integer order;
    private List<MenuVM> subMenu;
	public MenuVM() {
	}
	public MenuVM(SysMenu sm) {
        BeanUtils.copyProperties(sm, this);
        this.stateRef = sm.getName();
    }
	
}

File: king-admin-kotlin/src/main/kotlin/com/oukingtim/web/vm/MenuVM.kt
package com.oukingtim.web.vm

import com.oukingtim.domain.SysMenu
import org.springframework.beans.BeanUtils

/**
 * Created by oukingtim
 */
class MenuVM() {
     var id: String? = null
     var parentId: String? = null
     var name: String? = null
     var stateRef: String? = null
     var type: String? = null
     var icon: String? = null
     var title: String? = null
     var level: Int? = null
     var order: Int? = null
     var subMenu: List<MenuVM>? = null
    constructor(sm : SysMenu):this(){
        this.stateRef = sm.name
        this.name = sm.name
        this.type = sm.type
        this.icon = sm.icon
        this.title = sm.title
        this.level = sm.level
        this.order = sm.order
        this.parentId = sm.parentId
        this.id = sm.id
    }
}